### PR TITLE
FEATURE: BB-643 Add headlines secondary measure

### PR DESCRIPTION
**Pull request by petrjanu**

### CODE REVIEW

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1036/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/830/ : SUCCESS in 1m 01s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/303/ : SUCCESS in 2m 09s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/293/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1037/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/293/ : UNSTABLE in 4m 18s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/294/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1038/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/294/ : UNSTABLE in 4m 23s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1039/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/831/ : SUCCESS in 55s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/304/ : SUCCESS in 2m 23s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/295/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1040/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/295/ : UNSTABLE in 4m 14s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1041/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/832/ : SUCCESS in 1m 03s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/305/ : SUCCESS in 2m 29s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1043/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/833/ : SUCCESS in 1m 04s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/306/ : SUCCESS in 2m 12s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/297/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1044/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/297/ : SUCCESS in 4m 19s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1045/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/834/ : SUCCESS in 2m 21s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/307/ : SUCCESS in 2m 43s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/298/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1046/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/298/ : SUCCESS in 5m 23s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1048/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/835/ : SUCCESS in 2m 34s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/308/ : SUCCESS in 2m 14s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/300/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1049/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/300/ : UNSTABLE in 4m 19s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1050/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/836/ : SUCCESS in 53s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/309/ : SUCCESS in 2m 11s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/301/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1051/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/301/ : SUCCESS in 4m 04s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1052/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/837/ : FAILURE in 2m 30s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/310/ : SUCCESS in 2m 42s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/302/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1053/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/302/ : UNSTABLE in 5m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1054/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/838/ : SUCCESS in 59s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/311/ : SUCCESS in 2m 08s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/303/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1055/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/303/ : UNSTABLE in 4m 10s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/304/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1056/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/304/ : SUCCESS in 3m 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1057/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/839/ : SUCCESS in 1m 05s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/312/ : SUCCESS in 2m 16s

File: src/components/afm/tests/utils/DummyHeadlineExecuteAfmAdapter.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// (C) 2007-2018 GoodData Corporation
+import { ISdk } from 'gooddata';
+import { DataSource, ExecuteAfmAdapter } from '@gooddata/data-layer';
+import { AFM, Execution } from '@gooddata/typings';
+import IDataSource = DataSource.IDataSource;
```
Comment (martinmosko) [inline src/components/afm/tests/utils/DummyHeadlineExecuteAfmAdapter.tsx]: it looks like a auto import

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1058/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/840/ : FAILURE in 1m 38s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/313/ : SUCCESS in 2m 39s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1074/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/308/ : FAILURE in 1m 46s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1073/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/852/ : SUCCESS in 6m 07s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/325/ : SUCCESS in 4m 04s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1075/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/309/ : SUCCESS in 5m 20s

File: src/components/afm/tests/utils/DummyHeadlineExecuteAfmAdapter.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        const dataSource = new DataSource.DataSource<Execution.IExecutionResponses>(
+            execFactory,
+            afm,
+            fingerprint
+        );
+        return Promise.resolve(dataSource);
+    }
+}
+
+export function dummyHeadlineExecuteAfmAdapterFactory(sdk: ISdk, projectId: string) {
```
Comment (martinmosko) [inline src/components/afm/tests/utils/DummyHeadlineExecuteAfmAdapter.tsx]: https://en.wikipedia.org/wiki/Factory_method_pattern please change the name its not factory by definition.

File: src/components/core/tests/Headline.spec.tsx [hunk 3]
```diff
@@ -12,34 +13,66 @@ import { HeadlineTransformation } from '@gooddata/indigo-visualizations';
...
+                    executionResponse: expect.any(Object),
+                    executionResult: expect.any(Object),
+                    onAfterRender: wrapperProps.afterRender,
+                    drillableItems: wrapperProps.drillableItems
+                });
             });
         });
     });
-});
+ });
```
Comment (martinmosko) [inline src/components/core/tests/Headline.spec.tsx]: whitespace?

File: src/components/tests/mocks.tsx [hunk 4]
```diff
@@ -45,6 +47,18 @@ export const oneMeasureDataSource: IDataSource = {
...
 
+export const oneMeasureOneDimensionDataSource: IDataSource = {
+    getData: () => Promise.resolve(oneMeasureOneDimensionResponse),
+    getAfm: () => oneMeasureAfm,
+    getFingerprint: () => JSON.stringify(oneMeasureOneDimensionResponse)
+};
+
+export const twoMeasuresOneDimensionDataSource: IDataSource = {
+    getData: () => Promise.resolve(twoMeasuresOneDimensionResponse),
+    getAfm: () => oneMeasureAfm,
```
Comment (martinmosko) [inline src/components/tests/mocks.tsx]: oneMeasureAfm ?

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1076/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/853/ : SUCCESS in 4m 11s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/326/ : SUCCESS in 2m 50s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/308/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1077/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/310/ : SUCCESS in 4m 28s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/200/ : SUCCESS in 30s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/854/ : SUCCESS in 1m 26s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/309/ : UNSTABLE in 6m 17s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/327/ : SUCCESS in 4m 30s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/201/ : SUCCESS in 18s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/855/ : SUCCESS in 1m 45s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/310/ : UNSTABLE in 4m 18s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/328/ : SUCCESS in 3m 08s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/202/ : SUCCESS in 2m 31s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/856/ : SUCCESS in 1m 21s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/311/ : UNSTABLE in 4m 26s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/329/ : SUCCESS in 2m 15s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/203/ : SUCCESS in 54s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/857/ : FAILURE in 1m 19s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/312/ : UNSTABLE in 5m 18s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/330/ : SUCCESS in 3m 43s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1078/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/858/ : SUCCESS in 1m 23s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/331/ : SUCCESS in 2m 56s

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/313/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1079/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/311/ : UNSTABLE in 4m 44s

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by osklive-kacatko** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/314/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/1080/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/312/ : SUCCESS in 5m 51s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/204/ : SUCCESS in 19s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/859/ : SUCCESS in 4m 08s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/315/ : SUCCESS in 4m 00s
- https://ci.intgdc.com/job/gooddata-react-components-examples-validation-zuul-docker-el7/332/ : SUCCESS in 4m 00s

**Pull request merged**